File: sicpa-web/src/app/models/enterprise-dto.ts
import { DepartmentDto } from "./department-dto";
import { GenericDto } from "./generic-dto";

export class EnterpriseDto extends GenericDto {

  id: number;
  address: string;
  name: string
  phone: string;
  //departments: Array<DepartmentDto>;


  constructor() {
    super();
  }
}

File: sicpa-root/sicpa-dto/src/main/java/com/sicpa/demo/dto/EnterpriseDto.java
package com.sicpa.demo.dto;

import java.io.Serializable;

import javax.validation.constraints.NotEmpty;

import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = true)
public class EnterpriseDto extends BaseDto implements Serializable {
	private static final long serialVersionUID = 1L;

	private Long id;
	@NotEmpty
	private String address;
	@NotEmpty
	private String name;
	private String phone;
	//private List<DepartmentDto> departments;

}
